pyuvdata/uvflag.py

```diff
@@ -156,45 +366,686 @@ def __init__(self, input, mode='metric', copy_flags=False, waterfall=False, hist
+        self._weights_array.form = ('Nblts', 'Nspws', 'Nfreqs', 'Npols')
+
+        desc = ('Array of unique times, center of integration, shape (Ntimes), '
+                'units Julian Date')
+        self._time_array.form = ('Nblts',)
+
+        desc = ('Array of unique lsts, center of integration, shape (Ntimes), '
+                'units radians')
+        self._lst_array.form = ('Nblts',)
+
+        desc = ('Array of frequencies, center of the channel, '
+                'shape (Nspws, Nfreqs), units Hz')
+        self._freq_array.form = ('Nspws', 'Nfreqs')
+
+    def _set_type_waterfall(self):
+        """Set the type and required propertis consistent with a waterfall object."""
+        self.type = 'waterfall'
+        self._ant_array.required = False
+        self._baseline_array.required = False
+        self._ant_1_array.required = False
+        self._ant_2_array.required = False
+        self._Nants_telescope.required = False
+        self._Nants_data.required = False
+        self._Nbls.required = False
+        self._Nspws.required = False
+        self._weights_array.required = True
 
-        array_list = ['weights_array', 'time_array', 'lst_array', 'freq_array',
-                      'polarization_array']
-        if self.type == 'antenna':
-            array_list += ['ant_array']
-        elif self.type == 'baseline':
-            array_list += ['baseline_array', 'ant_1_array', 'ant_2_array', 'Nants_telescope']
-        if self.mode == 'flag':
-            array_list += ['flag_array']
-        elif self.mode == 'metric':
-            array_list += ['metric_array']
-        for arr in array_list:
-            self_param = getattr(self, arr)
-            other_param = getattr(other, arr)
-            if not np.all(self_param == other_param):
-                return False
+        self.Nblts = self.Ntimes
+
+        desc = 'Floating point metric information, shape (Nblts, Nfreqs, Npols).'
+        self._metric_array.desc = desc
+        self._metric_array.form = ('Nblts', 'Nfreqs', 'Npols')
+
+        desc = 'Boolean flag, True is flagged, shape (Nblts, Nfreqs, Npols)'
+        self._flag_array.desc = desc
+        self._flag_array.form = ('Nblts', 'Nfreqs', 'Npols')
+
+        desc = 'Floating point weight information, has shape (Nblts, Nfreqs, Npols).'
+        self._weights_array.desc = desc
+        self._weights_array.form = ('Nblts', 'Nfreqs', 'Npols')
+
+        desc = ('Array of unique times, center of integration, shape (Ntimes), '
+                'units Julian Date')
+        self._time_array.form = ('Ntimes',)
+
+        desc = ('Array of unique lsts, center of integration, shape (Ntimes), '
+                'units radians')
+        self._lst_array.form = ('Ntimes',)
+
+        desc = ('Array of frequencies, center of the channel, '
+                'shape (Nfreqs), units Hz')
+        self._freq_array.form = ('Nfreqs',)
 
+    def __eq__(self, other, check_history=False, check_extra=False):
```

Comment:
I'm surprised that the check_history option is defaulted to False. Do you generally (not just for testing) expect them not to match?

Reply:
I think @adampbeardsley might be able to provide insight on this. Maybe it was false before since the history was a bit of a hack with all the version strings.

Reply:
for what it is worth, I can set these options both to true and all tests still pass locally

Reply:
I'd vote to set both of them to `True` by default.